Tan Eu Zin iP

euzintan opened the pull request.

Commit:
Add Gradle support

euzintan committed:
This is level 1, I hope im committing this correctly.

euzintan committed:
Level 2

euzintan committed:
Level 2

euzintan committed:
Revert "Level 2"

This reverts commit 193dc5bcbe56349d85005e56b66ff362c456a721.

euzintan committed:
Level 3

euzintan committed:
Level 4:
Added classes Event and Deadline
Changed response to print via toString()
Added a counter that tells users number of things in list

euzintan committed:
Merge remote-tracking branch 'origin/master' into master

euzintan committed:
Added Automated Text UI Testing

euzintan committed:
Implemented Duke specific extensions

euzintan committed:
Implemented Delete

euzintan committed:
Updated Exceptions to continue accepting input after exceptions thrown

BILLXYR commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
-        System.out.println("Hello from\n" + logo);
+        System.out.println(logo + "\nHello im Eu Zin's Duke, he spent thursday afternoon creating me cuz he forgot abt the iP");
+
+        ArrayList<Task> taskList = new ArrayList<>();
```
From the websites it says that : "Plural form should be used on names representing a collection of objects."
So I think perhaps it should be taskLists instead of taskList.

BILLXYR commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+public class Deadline extends Task {
```
the websites says: "Class variables should never be declared public"
So, perhaps it should not public.

BILLXYR commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+import java.util.Iterator;
+import java.util.Scanner;
+
 public class Duke {
```
the websites says: "Class variables should never be declared public"
So, perhaps it should not public.

BILLXYR commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+public class DukeException extends Throwable {
```
the websites says: "Class variables should never be declared public"
So, perhaps it should not public.

BILLXYR commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,18 @@
+public class Event extends Task{
```
the websites says: "Class variables should never be declared public"
So, perhaps it should not public.

BILLXYR commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+public class Task {
```
the websites says: "Class variables should never be declared public"
So, perhaps it should not public.

BILLXYR reviewed (commented):
Two main problems observed: "the websites says: "Class variables should never be declared public"
So, perhaps it should not public." and " Plural form should be used on names representing a collection of objects." Others are fine 👍  Good job 👍

euzintan committed:
Created new class Storage to deal with saving and loading
Implemented saving to and laoding from hard disk function

euzintan committed:
Created new class Storage to deal with saving and loading
Implemented saving to and laoding from hard disk function

euzintan committed:
Changed Deadline class so Duke can deal with LocalDate and LocalTime rather than string

euzintan committed:
Merge branch 'branch-Level-8'

euzintan committed:
Resolved conflicts caused during merging of save to harddisk feature and recognising LocalDate feature

euzintan committed:
Created Parser and Ui classes

euzintan committed:
Added JUnit Tests for Parser

euzintan committed:
Added JavaDoc Comments

euzintan committed:
Implemented Find feature

euzintan committed:
Made changes to comply to Coding Standards

euzintan committed:
Resolved conflicts when merging increment branches

# Conflicts:
#	src/main/java/TaskList.java
#	src/main/java/Ui.java

euzintan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

juandavinlie commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+                boolean isDone = taskDataDivided[2].equals("1");
+                switch (taskDataDivided[0]) {
+                    case "E":
+                        returnTaskList.add(new Event(taskDataDivided[2], taskDataDivided[3], isDone));
```
Should taskDataDivided[0, 2, 3] be assigned to a variable that is self-descriptive? It would make the code easier to read and understand.

juandavinlie commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+     * @return copy of ArrayList
+     */
+    public ArrayList<Task> getList() {
+        ArrayList<Task> returnTaskList = new ArrayList<>();
```
It might be more appropriate to name this variable as taskList instead of returnTaskList since return is a verb and could refer to a function.

juandavinlie commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+    }
+
+    /**
+     * Add Task given to the list
+     * @param task
```
It is good to separate the description and parameter using a blank line.

juandavinlie commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+    public ArrayList<Task> searchFor(String searchString) {
+        ArrayList<Task> returnArrayList = new ArrayList<>();
+        for (Task task : taskList) {
+            if (task.getDescription().contains(searchString)) returnArrayList.add(task);
```
It may be better to use the Egyptian style when writing if-else statement as it becomes easier to read.

juandavinlie commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+        return taskList.size();
+    }
+
+    public ArrayList<Task> searchFor(String searchString) {
```
May be better to name it searchTaskWithName(String searchString), as it is understandable by just reading it.

juandavinlie reviewed (commented):
Overall, your code is well written and understandable. Some names of variables and methods can be modified to make it self-explanatory. Good job!!

euzintan committed:
Prepared Duke for GUI, tested out Hello World

euzintan committed:
Created Working GUI for Duke
All features of Duke available on GUI

euzintan committed:
Added assertions in Parser class to make sure input of the user is appropriate before any processing

euzintan committed:
Improve Code Quality

euzintan committed:
Created shadowJar file

euzintan committed:
Merge pull request #1 from euzintan/branch-Assertions

Pull Request for Assertions

euzintan committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java

euzintan committed:
Implemented FixedDurationTasks so that durations can be added to all types of tasks

euzintan committed:
SLAPed harder to reduce abstraction problems

Made Exceptions more explicit so the user can know how to enter an appropriate command.

euzintan committed:
Set theme jekyll-theme-slate

euzintan committed:
Edited README.md to be a proper User Guide

euzintan committed:
Merge from GitHub to change theme

euzintan committed:
Improved User Guide

Added Product showcase screenshot

euzintan committed:
Changed Ui Screenshot

euzintan committed:
Fixed Duke to work with Mac
